test/integration_test.py

```diff
@@ -148,11 +148,21 @@ def test(self):
         if config.deployment_stage != 'prod':
             try:
                 self._test_indexing()
+            except Exception as e:
+                log.error('Exception potentially due to timeout expiring while'
+                          'waiting for queues to drain. Cleaning up ...', exc_info=e)
+                self._delete_bundles(self.test_notifications)
+            else:
                 self._test_manifest()
                 if config.dss_direct_access:
                     self._test_dos_and_drs()
+                self._delete_bundles(self.test_notifications)
+                # Delete some bundles again to test that we handle duplicate deletions.
+                # Note: random.choices() may pick the same element multiple times so
+                # some notifications will end up being sent three or more times.
+                notifications = random.choices(self.test_notifications, k=len(self.test_notifications) // 2)
+                self._delete_bundles(notifications)
             finally:
```

Comment:
I'm pretty sure I we agreed to **replace** `finally` with `except` and `else`. Once an exception occurs, it is pointless to try to assert anything about the index.